benchmark/benchmark.php

```diff
@@ -0,0 +1,125 @@
+use Symfony\Component\Process\Process;
+use Symfony\Component\Process\Exception\ProcessTimedOutException;
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+error_reporting(E_ALL);
+chdir(dirname(__DIR__));
+
+function main(bool $commitResult) {
+    $data = [];
+    // $data['bench.php'] = runBench();
+    $data['symfony_demo'] = runSymfonyDemo();
+    $result = json_encode($data, JSON_PRETTY_PRINT) . "\n";
+
+    if ($commitResult) {
+        commitResult($result);
+    } else {
+        echo $result;
+    }
+}
+
+function commitResult(string $result) {
+    $repo = getDataRepo();
+    $repo->pull('origin');
+
+    $commitHash = getPhpSrcCommitHash();
+    $dir = __DIR__ . '/data/' . substr($commitHash, 0, 2) . '/' . $commitHash;
+    $summaryFile = $dir . '/summary.json';
+    if (!is_dir($dir)) {
+        mkdir($dir, 0755, true);
+    }
+    file_put_contents($summaryFile, $result);
+
+    $repo->addFile($summaryFile);
+    $repo->commit('Add result for php/php-src@' . $commitHash, ['--allow-empty']);
+    $repo->push();
+}
+
+function getPhpSrcCommitHash(): string {
+    $git = new Git();
+    $repo = $git->open(dirname(__DIR__));
+    return $repo->getLastCommitId();
+}
+
+function getDataRepo(): GitRepository {
+    $git = new CzProject\GitPhp\Git();
+    if (!is_dir(__DIR__ . '/data')) {
+        $repo = $git->cloneRepository('git@github.com:iluuu1994/php-benchmark-data.git', __DIR__ . '/data');
+    }
+    return $git->open(__DIR__ . '/data');
+}
+
+function runBench(): array {
+    $process = runValgrindPhpCommand('Zend/bench.php');
+    return ['instructions' => extractInstructionsFromValgrindOutput($process->getErrorOutput())];
+}
+
+function runSymfonyDemo(): array {
+    $git = new CzProject\GitPhp\Git();
+    if (!is_dir(__DIR__ . '/projects/symfony-demo')) {
+        $repo = $git->cloneRepository('git@github.com:iluuu1994/symfony-demo-2.2.3.git', __DIR__ . '/projects/symfony-demo');
+    }
+    runPhpCommand(__DIR__ . '/projects/symfony-demo/bin/console cache:clear');
+    runPhpCommand(__DIR__ . '/projects/symfony-demo/bin/console cache:warmup');
+    $process = runValgrindPhpCommand(__DIR__ . '/projects/symfony-demo/public/index.php');
+    return ['instructions' => extractInstructionsFromValgrindOutput($process->getErrorOutput())];
+}
+
+function runPhpCommand(string $args, ?int $timeout = null): Process {
+    $php = escapeshellarg(PHP_BINARY);
+    return runCommand("$php $args", $timeout);
+}
+
+function runValgrindPhpCommand(string $args, ?int $timeout = null): Process {
+    $php = escapeshellarg(PHP_BINARY);
+    return runValgrindCommand("$php $args", $timeout);
+}
+
+function runValgrindCommand(string $command, ?int $timeout = null): Process {
+    return runCommand("valgrind --tool=callgrind --dump-instr=yes --callgrind-out-file=/dev/null -- $command");
+}
+
+function runCommand(string $command, ?int $timeout = null): Process {
+    $process = Process::fromShellCommandline($command);
```

Comment:
Do you really need to pull symfony Process for this? Why don't you just use `shell_exec`? It would save a little bit of installation time...

Reply:
it would make sense if timeout was used but don't see it used anywhere...